MKS - Expose your applications using OVHcloud Public Cloud Load Balancer (#5518)
ovh/docs: pages/public_cloud/containers_orchestration/managed_kubernetes/expose_your_services_using_a_load_balancer/guide.en-gb.md

---
title: Expose your services using OVHcloud Public Cloud Load Balancer
excerpt: ""
updated: 2024-01-09
---

> [!warning]
>
> Usage of the [Public Load Balancer](https://www.ovhcloud.com/en-ie/public-cloud/load-balancer/) with Managed Kubernetes Service (MKS) is currently in Beta phase. This feature is also available to customers using the services of our US subsidiary.
>

## Objective

This guide aim to explain how to use our Public Cloud Load Balancer to expose your app hosted on our [Managed Kubernetes Service (MKS)](https://www.ovhcloud.com/en/public-cloud/kubernetes/).

> If you're not comfortable with the different ways of exposing your applications in Kubernetes, or if you're not familiar with the notion of service type 'loadbalancer', we recommend you start by reading [this guide](https://help.ovhcloud.com/csm/en-ie-public-cloud-kubernetes-using-lb?id=kb_article_view&sysparm_article=KB0050008) detailing the different methods for exposing your containerized applications hosted in Kubernetes.

Reviewer: ```suggestion
If you're not comfortable with the different ways of exposing your applications in Kubernetes, or if you're not familiar with the notion of Service's type 'LoadBalancer', we recommend you to start by reading [this guide](https://help.ovhcloud.com/csm/en-ie-public-cloud-kubernetes-using-lb?id=kb_article_view&sysparm_article=KB0050008) detailing the different methods for exposing your containerized applications hosted in Kubernetes.
```

> Our Public Cloud Load Balancer is relying on Openstack Octavia project, this project provides a Cloud Controller Manager (CCM) allowing Kubernetes clusters to interact with Load Balancers. For Managed Kubernetes Service (MKS), this Cloud Controller is installed and configured by our team allowing you to easily create, use and configure our Public Cloud Load Balancers. You can find the CCM opensource project documentation [here](https://github.com/kubernetes/cloud-provider-openstack/blob/master/docs/openstack-cloud-controller-manager/expose-applications-using-loadbalancer-type-service.md)

Reviewer: ```suggestion
Our Public Cloud Load Balancer is relying on Openstack Octavia project, this project provides a Cloud Controller Manager (CCM) allowing Kubernetes clusters to interact with Load Balancers. For Managed Kubernetes Service (MKS), this Cloud Controller is installed and configured by our team allowing you to easily create, use and configure our Public Cloud Load Balancers. You can find the [CCM open source project documentation](https://github.com/kubernetes/cloud-provider-openstack/blob/master/docs/openstack-cloud-controller-manager/expose-applications-using-loadbalancer-type-service.md).
```

> This guide uses some concepts that are specific to our Public Cloud Load Balancer (listener, pool, health monitor, member, ...)  and to the Public Cloud Network (Gateway, Floating IP). You can find concepts description on this [page](../../../public_cloud_network_services/concepts-03-loadbalancer/guide.en-gb.md) for the Load Balancer and on that [page](../../../public_cloud_network_services/concepts-01-public-cloud-networking-concepts/guide.en-gb.md) for the Public Cloud Network products.

Reviewer: ```suggestion
This guide uses some concepts that are specific to our Public Cloud Load Balancer (listener, pool, health monitor, member, ...) and to the Public Cloud Network (Gateway, Floating IP). You can find [concepts description for the Load Balancer](../../../public_cloud_network_services/concepts-03-loadbalancer/guide.en-gb.md) and [concepts description for the Public Cloud Network products](../../../public_cloud_network_services/concepts-01-public-cloud-networking-concepts/guide.en-gb.md).
```

## Prerequisites

#### Kubernetes version
- To be able to deploy Public Cloud Load Balancers, your Managed Kubernetes Service must run or have been upgraded to the following patch versions:

| Kubernetes versions |
|-------------|
| 1.24.13-3>= |   
| 1.25.9-3>=  |   
| 1.26.4-3>=  |   
| 1.27>=      |  

> #### Network topologie to expose your Load Balancers publicly   //à reformuler

Reviewer: ```suggestion
#### Network prerequisite to expose your Load Balancers publicly
```

> If you plan to expose your Load Balancer publicly, in order to attach a FloatingIP to your Load Balancer, an OVHcloud Gateway (an Openstack router) is mandatory on the subnet hosting your Load Balancer.

Reviewer: ```suggestion
If you plan to expose your Load Balancer publicly, in order to attach a FloatingIP to your Load Balancer, an OVHcloud Gateway (an OpenStack router) is mandatory on the subnet hosting your Load Balancer.
```

> If you plan to expose your Load Balancer publicly, in order to attach a FloatingIP to your Load Balancer, an OVHcloud Gateway (an Openstack router) is mandatory on the subnet hosting your Load Balancer.

Reviewer: on a une doc sur le FloatingIP ?

car si l'utilisateur arrive ici et ne sait pas ce que c'est, il risque apres de decrocher sa lecture car il n'a pa compris le terme

Reviewer 2: yes on peut au moins lier ça en EN: https://www.ovhcloud.com/fr/public-cloud/floating-ip/

> If you plan to expose your Load Balancer publicly, in order to attach a FloatingIP to your Load Balancer, an OVHcloud Gateway (an Openstack router) is mandatory on the subnet hosting your Load Balancer.

Reviewer 2: ```suggestion
If you plan to expose your Load Balancer publicly, in order to attach a [FloatingIP](https://www.ovhcloud.com/en-ie/public-cloud/floating-ip/) to your Load Balancer, an OVHcloud Gateway (an Openstack router) is mandatory on the subnet hosting your Load Balancer.
```

> For such use case we do recommend to create your MKS clusters on a network and subnet using our OVHcloud Gateway.

Reviewer: "our" ou "your" ?

Reviewer 2: 'your' me semble ok

For existing clusters, if:

> - **The Subnet's GatewayIP is already an OVHcloud Gateway**, nothing needs to be done. The current Openstack Router will be used.

Reviewer: ```suggestion
- **The Subnet's GatewayIP is already an OVHcloud Gateway**, nothing needs to be done. The current OpenStack Router will be used.
```

> - **The subnet doest not have an IP reserved for a Gateway** --> You will have to provide/create a compatible subnet. You can edit the existing subnet (modop CLI Openstack/Horizon) or use another dedicated subnet (doc LoadBalancerSubnetID)//TODO

Reviewer: ```suggestion
- **The subnet does not have an IP reserved for a Gateway** --> You will have to provide/create a compatible subnet. You can edit the existing subnet (modop CLI Openstack/Horizon) or use another dedicated subnet (doc LoadBalancerSubnetID)//TODO
```

> - **The subnet doest not have an IP reserved for a Gateway** --> You will have to provide/create a compatible subnet. You can edit the existing subnet (modop CLI Openstack/Horizon) or use another dedicated subnet (doc LoadBalancerSubnetID)//TODO

Reviewer: modop?

Reviewer 2: mode opératoire, c'est du reste à faire

- **The GatewayIP is already assigned to a non-OVHcloud Gateway (Openstack Router)** --> You will have to provide/create a compatible subnet. To do so you select another dedicated subnet (doc LoadBalancerSubnetID)//TODO

## Limitations

- Layer 7 Policy & Rules and TLS Termination (`TERMINATED_HTTPS` listener) are not available yet. For such use cases you can rely on [Octavia Ingress Controller](https://github.com/kubernetes/cloud-provider-openstack/blob/master/docs/octavia-ingress-controller/using-octavia-ingress-controller.md)
- UDP proxy protocol is not supported

## Billing

> - When exposing your load balancer publicly (pub-to-pub or pub-to-private):

Reviewer: ```suggestion
- When exposing your load balancer publicly (public-to-public or public-to-private):
```

  - if it does not already exist, an OVHcloud Gateway is automatically created and charged for all Load Balancers spawned in the subnet <https://www.ovhcloud.com/en-gb/public-cloud/prices/#10394>.
  - a Public Floating IP will be used: https://www.ovhcloud.com/en-gb/public-cloud/prices/#10346

> - Each Public CLoud Load Balancer is billed according to his flavor: <https://www.ovhcloud.com/en-gb/public-cloud/prices/#10420>

Reviewer: ```suggestion
- Each Public Cloud Load Balancer is billed according to his flavor: <https://www.ovhcloud.com/en-gb/public-cloud/prices/#10420>
```

    > [!primary]
    >
    > Note: Each publicly exposed Load Balancer has his own Public Floating IP. Outgoing traffic doesn't consume OVHcloud Gateway bandwidth.
    >

    > [!warning]
    >

>     > During the CCM Beta, since the Public Cloud Load Balancer is GA, the Public Cloud Load Balancer usage as well as the other network components (Gateway & Floating IPs) will be billed

Reviewer 2: ```suggestion
    > During the Public Cloud Load Balancer with Manage Kubernetes Service Beta, since the Public Cloud Load Balancer is GA, the Load Balancer usages as well as the other network components (Gateway & Floating IPs) will be charged
```

    >

> ## Process

Reviewer: Instructions ?

> ## Process

Reviewer: ```suggestion
## Instructions
```

> During the beta phase, if you want a Kubernetes load balancer service to be deployed using [Public Cloud Load Balancer](https://www.ovhcloud.com/en-ie/public-cloud/load-balancer/) rather than the current [Loadbalancer for Kubernetes](https://www.ovhcloud.com/en-ie/public-cloud/load-balancer-kubernetes/) solution, you'll need to add an annotation: loadbalancer.ovhcloud.com/class: "octavia" on your Kubernetes Service manifest.

Reviewer: ```suggestion
During the Beta phase, if you want a Kubernetes Load Balancer service to be deployed using [Public Cloud Load Balancer](https://www.ovhcloud.com/en-ie/public-cloud/load-balancer/) rather than the current [Loadbalancer for Kubernetes](https://www.ovhcloud.com/en-ie/public-cloud/load-balancer-kubernetes/) solution, you'll need to add an annotation: `loadbalancer.ovhcloud.com/class: "octavia"` on your Kubernetes Service manifest.
```

> Here's a simple example of how to use the Public Cloud Load Balancer

Reviewer: ```suggestion
Here's a simple example of how to use the Public Cloud Load Balancer:
```

> 1. Creation of a functional Managed Kubernetes (MKS) cluster using the OVHcloud manager, Terraform or APIs.

Reviewer: ajouter les liens des docs qui expliquent via le manager, via terraform, via pulumi, via l'api ... :)

(quand on ecrit un tutoriel il faut toujours donner toutes les infos a l'utilisateur)
(si l'utilisateur doit chercher un lien, d'une il n'a pas à le faire car nous on le connait et de deux il risque de ne pas tomber sur le bon, sur un site externe, sur une doc fausse, pas a jour ... :) )

> 1. Creation of a functional Managed Kubernetes (MKS) cluster using the OVHcloud manager, Terraform or APIs.

Reviewer: ```suggestion
1. Deployment of a functional Managed Kubernetes (MKS) cluster using the OVHcloud manager, Terraform or APIs.
```

> 1. Creation of a functional Managed Kubernetes (MKS) cluster using the OVHcloud manager, Terraform or APIs.

Reviewer 2: ```suggestion
1. Creation of a functional Managed Kubernetes (MKS) cluster using the [OVHcloud manager](https://help.ovhcloud.com/csm/en-ie-public-cloud-kubernetes-create-cluster?id=kb_article_view&sysparm_article=KB0037221), [Terraform](https://help.ovhcloud.com/csm/en-ie-public-cloud-kubernetes-create-cluster-with-terraform?id=kb_article_view&sysparm_article=KB0049684) or [APIs](https://api.ovh.com/console/#/cloud/project/{serviceName}/kube~POST).
```

> 1. Creation of a functional Managed Kubernetes (MKS) cluster using the OVHcloud manager, Terraform or APIs.

Reviewer 2: ```suggestion
1. Creation of a functional Managed Kubernetes (MKS) cluster using the [OVHcloud manager](https://help.ovhcloud.com/csm/en-ie-public-cloud-kubernetes-create-cluster?id=kb_article_view&sysparm_article=KB0037221), [Terraform](https://help.ovhcloud.com/csm/en-ie-public-cloud-kubernetes-create-cluster-with-terraform?id=kb_article_view&sysparm_article=KB0049684) or [APIs](https://api.ovh.com/console/#/cloud/project/{serviceName}/kube~POST).
```

> 2. Retrieve the kubeconfig needed to use kubectl tool (via OVHcloud manager, Terraform or API). You can use [this guide](https://help.ovhcloud.com/csm/en-ie-public-cloud-kubernetes-configure-kubectl?id=kb_article_view&sysparm_article=KB0049658)

Reviewer: ```suggestion
2. Retrieve the kubeconfig file needed to use `kubectl` tool (via OVHcloud manager, Terraform, Pulumi or API). You can use [this guide](https://help.ovhcloud.com/csm/en-ie-public-cloud-kubernetes-configure-kubectl?id=kb_article_view&sysparm_article=KB0049658).
```

> 3. Create a 'deployment' using the following command:

Reviewer: ```suggestion
3. Create a 'deployment' resource using the following command:
```

```shell
kubectl create deployment test-octavia --image=nginx
```
4. Copy/Paste the following code on a file named `test-lb-todel.yaml`
```yaml
apiVersion: v1
kind: Service
metadata:
  labels:
    app: test-octavia
  name: test-lb-todel
  namespace: default
  annotations:
    loadbalancer.ovhcloud.com/class: "octavia"
    loadbalancer.ovhcloud.com/flavor: "small"
spec:
  ports:
  - name: 80-80
    port: 80
    protocol: TCP
    targetPort: 80
  selector:
    app: test-octavia
  type: LoadBalancer
```

> 6. Create a 'service' using the following command:

Reviewer: ```suggestion
6. Create a 'Service' using the following command:
```

```shell

> kubectl create service test-lb-todel -f test-lb-todel.yaml

Reviewer: Dans tous les autres guides on utilise la commande `kubectl apply` pour appliquer un YAML.

> kubectl create service test-lb-todel -f test-lb-todel.yaml

Reviewer: Dois t'on deployer le service dans un namespace particulier ?

Si non, alors en ne fournissant pas de namespace, il faudrait préciser dans les commandes suivantes "-n default".

Bonne pratique : ne rien deployer dans le ns default :)

> kubectl create service test-lb-todel -f test-lb-todel.yaml

Reviewer 2: ```suggestion
kubectl apply -f test-lb-todel.yaml
```

```

> 7. Retrieve service IP address using the following command line:

Reviewer: ```suggestion
7. Retrieve Service IP address using the following command line:
```

```shell

> $ kubectl get service test-lb-todel

Reviewer: preciser le namespace

NAME                 TYPE           CLUSTER-IP    EXTERNAL-IP      PORT(S)        AGE
test-lb-todel   LoadBalancer   10.3.107.18   141.94.215.240   80:30172/TCP   12m
```

> 8. Open a web browser and access: http://IP-retrieved-in-prevous-step

Reviewer: ```suggestion
8. Open a web browser and access to: http://IP-retrieved-in-prevous-step
```

## Use cases

> 

Reviewer: le YAML au dessus fait reference a un plan 'small'.

Peux t'on indiquer dans ce guide les différentes valeurs que peux avoir ce plan, et a quoi cela correspond ?

Reviewer 2: c'est indiqué dans la liste des annotations

You can find a set a examples on how to use our Public Cloud Load Balancer with Managed Kubernetes Service (MKS) on our dedicated Github repository https://github.com/ovh/public-cloud-examples

#### Public-to-Public (you are using a public Managed Kubernetes Cluster)
//TODO

#### Public to Private (your cluster is attached to a private network/subnet)
In a public-to-private scenario you will use your Load Balancer to publicly expose app that are hosted on your Managed Kubernetes Cluster. Main benefit is that your Kubernetes nodes are not exposed on internet with that scenario.

Service example:
```yaml
apiVersion: v1
kind: Service
metadata:
  name: my-lb-service
  annotations:
    loadbalancer.ovhcloud.com/class: "octavia"

>     loadbalancer.ovhcloud.com/flavor: "medium"

Reviewer: La flavor medium est obligatoire ?
non fonctionnel en small ?
Si c'est necessaire alors je pense que ça serait utile de dire pourquoi et quel incidence coté billing ou autre, non ?

Reviewer 2: indiqué dans la liste des flavors (en annotation)

  labels:
    app: test-octavia
spec:
  ports:
  - name: client
    port: 80
    protocol: TCP
    targetPort: 80
  selector:
    app: nginx
  type: LoadBalancer
```

#### Private to Private
In a private-to-private scenario your Load Balancer is not exposed publicly, it might be useful if you want to expose your containerized service inside you OVHcloud private network.

Service example:
```yaml
apiVersion: v1
kind: Service
metadata:
  name: my-lb-service
  annotations:

>     loadbalancer.ovhcloud.com/class: "octavia"

Reviewer: pas de plan donc small par defaut ?

Reviewer 2: idem

  Member subnet ID of the load balancer created. By default, the subnet-id of the subnet configured for your OVHcloud Managed Kubernetes Service cluster will be used.

- `loadbalancer.openstack.org/network-id`

  The network ID which will allocate virtual IP for loadbalancer. By default, the network-id of the network configured for your OVHcloud Managed Kubernetes Service cluster will be used.

- `loadbalancer.openstack.org/port-id`

  The port ID for load balancer private IP. Can be used if you want to use a specific private IP.

- `loadbalancer.openstack.org/connection-limit`

  The maximum number of connections per second allowed for the listener. Positive integer or -1 for unlimited (default). This annotation supports update operation.

- `loadbalancer.openstack.org/keep-floatingip`

  If 'true', the floating IP will **NOT** be deleted upon load balancer deletion. Default is 'false'. Useful if you want to keep your floating API after Load Balancer deletion.

- `loadbalancer.openstack.org/proxy-protocol`

  If 'true', the loadbalancer pool protocol will be set as `PROXY`. Default is 'false'.

- `loadbalancer.openstack.org/x-forwarded-for`

>   **Not supported**. If you want perform Layer 7 load balancing we do recommend to use the official Octavia Ingress-controller: https://github.com/kubernetes/cloud-provider-openstack/blob/master/docs/octavia-ingress-controller/using-octavia-ingress-controller.md  

Reviewer: perform?

Reviewer 2: yes ?

- `loadbalancer.openstack.org/default-tls-container-ref` // NOT SUPPORTED

  Reference to a tls container. This option works with Octavia, when this option is set then the cloud provider will create an Octavia Listener of type `TERMINATED_HTTPS` for a TLS Terminated loadbalancer.
  Format for tls container ref: `https://{keymanager_host}/v1/containers/{uuid}`

  When `container-store` parameter is set to `external` format for `default-tls-container-ref` could be any string.

### Features
#### Resize your LoadBalancer //TODO (sample on github)
There is no proper way to 'resize' your loadbalancer yet (work in progress). The best alternative to change the flavor, is to recreate a new Kubernetes Service that will use the same public IP than the previous one.
You can find the HowTo and example on our public Github: https://github.com/ovh/public-cloud-examples

- First, make sure that the existing service is using the `loadbalancer.openstack.org/keep-floatingip` annotation, if not the public floating IP will be released. (It can be added after the service creation).
- Get the public IP of your existing service
```shell

> $ kubectl get service my-small-lb

Reviewer: preciser le namespace

NAME                 TYPE           CLUSTER-IP    EXTERNAL-IP      PORT(S)        AGE
test-lb-todel        LoadBalancer   10.3.107.18   141.94.215.240   80:30172/TCP   12m
```

> - Create a new service  with the new expected flavor:

Reviewer: ```suggestion
- Create a new service with the new expected flavor:
```

  ```yaml
  apiVersion: v1
  kind: Service
  metadata:
    name: my-medium-lb
    annotations:
      loadbalancer.ovhcloud.com/class: "octavia"
      loadbalancer.ovhcloud.com/flavor: "medium"
    labels:
      app: demo-upgrade
  spec:

>     loadBalancerIP: 141.94.215.240 # Use the IP address from the previous service

Reviewer: on recupere comment cette IP ? :)

Reviewer 2: c'est indiqué juste au dessus. Pas clair ?

    ports:
    - name: client
      port: 80
      protocol: TCP
      targetPort: 80
    selector:
      app: nginx
    type: LoadBalancer

>     ```

Reviewer: Ajouter la commande kubectl apply -f .. sinon le service ne sera pas créé

- Until the deletion of the previous service, this Service will only deploy the LoadBalancer without a floating IP.

> - When the Floating IP is available (the deletion of the initial LB service will unbound the IP), the floating ip will be attach to this new LB.

Reviewer: ```suggestion
- When the Floating IP is available (the deletion of the initial LB service will unbound the IP), the floating IP will be attach to this new LB.
```

> [!warning]
>

> > Changing the flavor will create some outage.

Reviewer: C'est a dire ?

Reviewer 2: Comme indiqué au dessus on delete/create le LB donc le temps de la bascule il y a une intéruption

Reviewer: Le terme "some outage" peut indiqué que l'utilisateur aura une panne, un probleme, cela a une connotation negative.
Plutot que d'utiliser le terme "some outage", dire : "Changing the flovor will create a new LB and delete the existing one so it might take some time"

un wording du genre ?

####  [Use PROXY protocol to preserve client IP](https://github.com/kubernetes/cloud-provider-openstack/blob/master/docs/openstack-cloud-controller-manager/expose-applications-using-loadbalancer-type-service.md#use-proxy-protocol-to-preserve-client-ip)

#### Migrate from LoadBalancer for Managed Kubernetes to Public Load Balancer //TODO
(Migration from IOLB to Octavia)
> [!warning]
>
> It is not possible to keep the existing public IP
>

- How to: change annotation lb/class from 'iolb' par 'Octavia'

## Ressources Naming //TODO , WIP
Explain how ressource created by LB are named.

## Others resources
- [Exposing applications using services of LoadBalancer type](https://github.com/kubernetes/cloud-provider-openstack/blob/master/docs/openstack-cloud-controller-manager/expose-applications-using-loadbalancer-type-service.md)
- [Using Octavia Ingress Controller](https://github.com/kubernetes/cloud-provider-openstack/blob/master/docs/octavia-ingress-controller/using-octavia-ingress-controller.md)
- [OVHcloud Load Balancer concept](https://help.ovhcloud.com/csm/en-gb-public-cloud-network-load-balancer-concepts?id=kb_article_view&sysparm_article=KB0059283)

## Go further

> Visit the [Github examples repository](https://github.com/ovh/public-cloud-databases-examples/tree/main/databases/cassandra) . //NOT EXISTING FOR MKS, TO BE CREATED

Reviewer: Si cela existe : https://github.com/ovh/public-cloud-examples/

Reviewer 2: Pas pour MKS, il faut que je m'occupe de la création du repo

Reviewer: Le repo que je t'ai indiqué liste plusieurs public cloud examples dont la creation d'un MKS :)

>

## Objective

This guide aim to know how to use our Public Cloud Load Balancer to expose your app hosted on our [Managed Kubernetes Service (MKS)](https://www.ovhcloud.com/en/public-cloud/kubernetes/).
If you're not comfortable with the different ways of exposing your applications in Kubernetes, or if you're not familiar with the notion of service type 'loadbalancer', we recommend you start by reading [this guide](https://help.ovhcloud.com/csm/en-ie-public-cloud-kubernetes-using-lb?id=kb_article_view&sysparm_article=KB0050008) detailing the different methods for exposing your containerized applications hosted in Kubernetes.

Our Public Cloud Load Balancer is relying on Openstack Octavia project, you can find the official documentation [here](https://github.com/kubernetes/cloud-provider-openstack/blob/master/docs/openstack-cloud-controller-manager/expose-applications-using-loadbalancer-type-service.md).

## Prerequisites

#### MKS Kubernetes version
- To be able to deploy Octavia load balancers, your Managed Kubernetes Service must have been upgraded to the following patch version:

> | Kubenertes versions    |

Reviewer: ```suggestion
| Kubernetes versions    |
```

    > During the CCM Beta, since the Public Cloud Load Balancer is GA, the Public Cloud Load Balancer usage as well as the other network components (Gateway & Floating IPs) will be billed
    >

## Instructions

> During the beta phase, if you want a Kubernetes load balancer service to be deployed using [Public Cloud Load Balancer](https://www.ovhcloud.com/en-ie/public-cloud/load-balancer/) rather than the historical [Loadbalancer for Kubernetes](https://www.ovhcloud.com/en-ie/public-cloud/load-balancer-kubernetes/) solution, you'll need to add the annotation: loadbalancer.ovhcloud.com/class: "octavia" on your Kubernetes Service manifest.

Reviewer: ```suggestion
During the beta phase, if you want a Kubernetes load balancer service to be deployed using [Public Cloud Load Balancer](https://www.ovhcloud.com/en-ie/public-cloud/load-balancer/) rather than the historical [Loadbalancer for Kubernetes](https://www.ovhcloud.com/en-ie/public-cloud/load-balancer-kubernetes/) solution, you'll need to add the annotation: `loadbalancer.ovhcloud.com/class: "octavia"` on your Kubernetes Service manifest.
```

Here's a simple example of how to use the Public Cloud Load Balancer

> 1. Deployment of a functional Managed Kubernetes (MKS) cluster using the [OVHcloud manager](https://help.ovhcloud.com/csm/en-ie-public-cloud-kubernetes-create-cluster?id=kb_article_view&sysparm_article=KB0037221),   [Terraform](https://help.ovhcloud.com/csm/en-ie-public-cloud-kubernetes-create-cluster-with-terraform?id=kb_article_view&sysparm_article=KB0049684), [Pulimi](https://help.ovhcloud.com/csm/en-ie-public-cloud-kubernetes-create-cluster-with-pulumi?id=kb_article_view&sysparm_article=KB0059712) or [APIs](https://api.ovh.com/console-preview/?section=%2Fcloud&branch=v1#post-/cloud/project/-serviceName-/kube).

Reviewer: ```suggestion
1. Deployment of a functional Managed Kubernetes (MKS) cluster using the [OVHcloud manager](https://help.ovhcloud.com/csm/en-ie-public-cloud-kubernetes-create-cluster?id=kb_article_view&sysparm_article=KB0037221), [Terraform](https://help.ovhcloud.com/csm/en-ie-public-cloud-kubernetes-create-cluster-with-terraform?id=kb_article_view&sysparm_article=KB0049684), [Pulumi](https://help.ovhcloud.com/csm/en-ie-public-cloud-kubernetes-create-cluster-with-pulumi?id=kb_article_view&sysparm_article=KB0059712) or [APIs](https://api.ovh.com/console-preview/?section=%2Fcloud&branch=v1#post-/cloud/project/-serviceName-/kube).
```

2. Retrieve the kubeconfig file needed to use kubectl tool (via OVHcloud manager, Terraform, Pulumi or API). You can use [this guide](https://help.ovhcloud.com/csm/en-ie-public-cloud-kubernetes-configure-kubectl?id=kb_article_view&sysparm_article=KB0049658)

> 3. Create a 'deployment' ressource using the following command:

Reviewer: ```suggestion
3. Create a Namespace and a Deployment resource using the following command:
```

```shell

> kunectl create Namespace test-lb-ns

Reviewer: ```suggestion
kunectl create namespace test-lb-ns
```

> kubectl create Deployment test-lb --image=nginx -n=test-lb-ns

Reviewer: ```suggestion
kubectl create deployment test-lb --image=nginx -n=test-lb-ns
```

    protocol: TCP
    targetPort: 80
  selector:
    app: test-lb
  type: LoadBalancer
```
6. Create a 'service' using the following command:
```shell
kubectl apply -f test-lb-service.yaml
```
7. Retrieve Service IP address using the following command line:
```shell
$ kubectl get service test-lb-service -n=test-lb-ns
NAME                 TYPE           CLUSTER-IP    EXTERNAL-IP      PORT(S)        AGE
test-lb-service      LoadBalancer   10.3.107.18   141.94.215.240   80:30172/TCP   12m
```

> 8. Open a web browser and access: http://141.94.215.240

Reviewer: warning to not display a running and working IP ;)

Reviewer 2: yep, fixed
